concurrent/src/main/scala/scalaz/concurrent/Future.scala

```diff
@@ -235,7 +264,7 @@ sealed abstract class Future[+A] {
 object Future {
   case class Now[+A](a: A) extends Future[A]
   case class Async[+A](onFinish: (A => Trampoline[Unit]) => Unit) extends Future[A]
-  case class Suspend[+A](thunk: () => Future[A]) extends Future[A]
+  case class Suspend[+A](thunk: () => Future[A], andFinally: Option[Option[Throwable] => Future[Unit]]) extends Future[A]
```

Comment:
I'm pretty sure the following will not free the resource:

```scala
val rsrc: Task[R] = ???
def free(r: R): Task[Unit] = ???

rsrc flatMap { r => now(r) onFinish { _ => free(r) } }
```

If `rsrc` is interrupted, I'm pretty sure the `onFinish` won't run at all.